jib-core/src/main/java/com/google/cloud/tools/jib/docker/CliDockerClient.java

```diff
@@ -246,11 +247,21 @@ public DockerImageDetails inspect(ImageReference imageReference)
       throws IOException, InterruptedException {
     Process inspectProcess =
         docker("inspect", "-f", "{{json .}}", "--type", "image", imageReference.toString());
-    if (inspectProcess.waitFor() != 0) {
-      throw new IOException(
-          "'docker inspect' command failed with error: " + getStderrOutput(inspectProcess));
+
+    try (InputStreamReader stdout =
+        new InputStreamReader(inspectProcess.getInputStream(), StandardCharsets.UTF_8)) {
+
+      String output = CharStreams.toString(stdout);
+
+      if (inspectProcess.waitFor() != 0) {
+        throw new IOException(
+            "'docker inspect' command failed with error: " + getStderrOutput(inspectProcess));
+      }
+
+      return JsonTemplateMapper.readJson(
+          new ByteArrayInputStream(output.getBytes(StandardCharsets.UTF_8)),
```

Comment:
Perhaps you can use [this](https://github.com/GoogleContainerTools/jib/blob/c2a43c5b786889f490b8d1314b5b0cd86bee3355/jib-core/src/main/java/com/google/cloud/tools/jib/json/JsonTemplateMapper.java#L118).

Reply:
Thanks for the tips. Updated the code to use the overload version you suggested.